main.js

```diff
@@ -72,9 +73,14 @@ const adapter = new utils.Adapter({
                 clearTimeout(timer);
             }
         });
-        // TODO: destroy rpi-gpio-buttons when this is fixed: https://github.com/bnielsen1965/rpi-gpio-buttons/issues/9
         if (gpio) {
-            gpio.destroy(() => callback && callback());
+            if (gpioButtons) {
+                gpioButtons.destroy().then(() => {
```

Comment:
also a catch making sense?

Reply:
Hmmmm... probably true. Will adapter still work if we make the `unload` function `async` or must it always do it's thing and use the `callback`?

Reply:
Don't worry - I looked in the code and answered this ;)

```javascript
                if (options.unload.length >= 1) {
                    // The method takes (at least) a callback
                    options.unload(finishUnload);
                } else {
                    // The method takes no arguments, so it must return a Promise
                    const unloadPromise = options.unload();
                    if (unloadPromise instanceof Promise) {
                        // Call finishUnload in the case of success and failure
                        unloadPromise.then(finishUnload, finishUnload);
                    } else {
                        // No callback accepted and no Promise returned - force unload
                        logger.error(`${this.namespaceLog} Error in ${id}: The unload method must return a Promise if it does not accept a callback!`);
                    }
                }
```